apps/meteor/app/api/server/lib/eraseTeam.ts

```diff
@@ -0,0 +1,101 @@
+import { AppEvents, Apps } from '@rocket.chat/apps';
+import { MeteorError, Team } from '@rocket.chat/core-services';
+import type { AtLeast, IRoom, ITeam, IUser } from '@rocket.chat/core-typings';
+import { Rooms, Users } from '@rocket.chat/models';
+
+import { eraseRoom } from '../../../../server/lib/eraseRoom';
+import { SystemLogger } from '../../../../server/lib/logger/system';
+import { deleteRoom } from '../../../lib/server/functions/deleteRoom';
+
+type eraseRoomFnType = (rid: string, userIdOrUser: string | AtLeast<IUser, '_id' | 'username' | 'name'>) => Promise<boolean | void>;
+
+export const eraseTeamShared = async (
+	userIdOrUserForUnset: string | AtLeast<IUser, '_id' | 'username' | 'name'>,
```

Comment:
Any particular reason for you to be allowing `string | IUser` here?

I think allowing just one of these might decrease complexity and maintenance... and also improve execution cost since you don't have to do a type check.